### PR TITLE
Rhosinfra 7 - Dynamic topology

### PR DESCRIPTION
**Pull request by obaranov**
@yfried-redhat @tkammer 
please review. poc gate passed: http://10.8.180.145:8080/job/poc-ir-obaranov-dynamic-topology/7/console

### CODE REVIEW

File: provision.example.ini [hunk 1]
```diff
@@ -5,7 +5,7 @@
 host=earth
 image-file=image.file.qcow2
 image-server=my.server.com
-extra-vars=[@/home/obaranov/local/redhat/git/khaleesi/khaleesi-settings/settings/product/rhos/private_settings/redhat_internal.yml]
+extra-vars=["@/home/obaranov/local/redhat/git/khaleesi/khaleesi-settings/settings/product/rhos/private_settings/redhat_internal.yml"]
```
Comment (yfried-redhat) [inline provision.example.ini]: Please remove this from public repo

File: cli/spec.py [hunk 2]
```diff
@@ -114,6 +114,38 @@ def override_default_values(clg_args, sub_parser_options):
...
+    if 'topology' in clg_args and clg_args['topology'] is not None:
+        topology_dict = {}
+        for topology_item in clg_args['topology'].split(','):
+            if '_' in topology_item:
+                number, node_type = topology_item.split('_')
+            else:
+                raise exceptions.IRConfigurationException(
+                    "Topology node should be in format  <number>_<node role>. "
+                    "Current value: '{}' ".format(topology_item))
+            if int(number) > 1 and node_type[-1] == 's':
```
Comment (yfried-redhat) [inline cli/spec.py]: This could invite bugs with english grammer :smile: 
Let's stick with matching the file name exactly.

File: README.rst [hunk 3]
```diff
@@ -107,6 +107,24 @@ option. There are 2 ways of doing so:
 The ``-e``/``--extra-vars`` can be used more than once.
 
 
+Dynamic Topology
+----------------
+
+InfraRed allows to dynamically define the provisioning topology to be used during deployment.
+
+InfraRed provides several 'mini' yam files to describe different roles: controller, compute, undercloud, etc.
```
Comment (aopincar) [inline README.rst]: typo: yam => yaml

File: cli/spec.py [hunk 4]
```diff
@@ -114,6 +114,36 @@ def override_default_values(clg_args, sub_parser_options):
...
+    Transforms some of arguments after they have been read.
+
+    :param clg_args: Dictionary based on cmd-line args parsed by clg
+    :param app_settings_dir: path to the base directory holding the
+        application's settings. App can be provisioner\installer\tester
+        and the path would be: settings/<app_name>/
+    """
+
+    # post process topology
+    if 'topology' in clg_args and clg_args['topology'] is not None:
```
Comment (aopincar) [inline cli/spec.py]: if clg_args.get('topology'):

File: cli/spec.py [hunk 5]
```diff
@@ -114,6 +114,36 @@ def override_default_values(clg_args, sub_parser_options):
         _check_required_arguments(sub_parser_options, clg_args)
 
 
+def post_process_command_args(clg_args, app_settings_dir):
```
Comment (aopincar) [inline cli/spec.py]: Please use more meaningful name for this method, it only deals with topology...

File: infrared.cfg.example [hunk 6]
```diff
@@ -5,4 +5,4 @@
 settings  = settings
 modules   = library
 roles     = roles
-playbooks = playbooks
+playbooks = playbooks
```
Comment (aopincar) [inline infrared.cfg.example]: Please add new line at the end of the file

File: settings/provisioner/topology/ceph.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+name: ceph
```
Comment (aopincar) [inline settings/provisioner/topology/ceph.yml]: Please add new line at the end of each file.

**Pull request merged**